### ISSUE TITLE
Passing RendererBuilder::build() error to app Layer specified does not exist

### ISSUE THREAD

**Issue by FlorianLudwig** [aclysma/skulpin#54]
The examples are not starting, error log:
```
[2020-07-14T07:42:17Z INFO  skulpin_renderer::instance] Finding vulkan entry point
[2020-07-14T07:42:17Z INFO  skulpin_renderer::instance] Found Vulkan version: (1, 2, 135)
[2020-07-14T07:42:17Z DEBUG skulpin_renderer::instance] Available Layers: [
    LayerProperties {
        layer_name: "VK_LAYER_MESA_device_select",
        spec_version: 4198473,
        implementation_version: 1,
        description: "Linux device selection layer",
    },
    LayerProperties {
        layer_name: "VK_LAYER_VALVE_steam_fossilize_32",
        spec_version: 4198473,
        implementation_version: 1,
        description: "Steam Pipeline Caching Layer",
    },
    LayerProperties {
        layer_name: "VK_LAYER_VALVE_steam_overlay_32",
        spec_version: 4198473,
        implementation_version: 1,
        description: "Steam Overlay Layer",
    },
    LayerProperties {
        layer_name: "VK_LAYER_VALVE_steam_fossilize_64",
        spec_version: 4198473,
        implementation_version: 1,
        description: "Steam Pipeline Caching Layer",
    },
    LayerProperties {
        layer_name: "VK_LAYER_VALVE_steam_overlay_64",
        spec_version: 4198473,
        implementation_version: 1,
        description: "Steam Overlay Layer",
    },
]
[2020-07-14T07:42:17Z DEBUG skulpin_renderer::instance] Available Extensions: [
    ExtensionProperties {
        extension_name: "VK_KHR_device_group_creation",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_external_fence_capabilities",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_external_memory_capabilities",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_external_semaphore_capabilities",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_get_display_properties2",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_get_physical_device_properties2",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_get_surface_capabilities2",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_surface",
        spec_version: 25,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_surface_protected_capabilities",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_wayland_surface",
        spec_version: 6,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_xcb_surface",
        spec_version: 6,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_xlib_surface",
        spec_version: 6,
    },
    ExtensionProperties {
        extension_name: "VK_KHR_display",
        spec_version: 23,
    },
    ExtensionProperties {
        extension_name: "VK_EXT_direct_mode_display",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_EXT_acquire_xlib_display",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_EXT_display_surface_counter",
        spec_version: 1,
    },
    ExtensionProperties {
        extension_name: "VK_EXT_debug_report",
        spec_version: 9,
    },
    ExtensionProperties {
        extension_name: "VK_EXT_debug_utils",
        spec_version: 1,
    },
]
[2020-07-14T07:42:17Z INFO  skulpin_renderer::instance] Creating vulkan instance
[2020-07-14T07:42:17Z WARN  skulpin_app_winit::app] Passing RendererBuilder::build() error to app Layer specified does not exist
Layer specified does not exist
```

I don't understand the error message :sweat:

**Comment by aclysma** [comment]
Most likely you do not have the vulkan SDK installed. This is required to run with validation enabled. (See the `.use_vulkan_debug_layer(true)` line in the example code. You could set this to false and maybe get it to run without needing the SDK. I recommend enabling this in debug mode for local development and disabling it in release/shipped code.

**Comment by FlorianLudwig** [comment]
Hi @aclysma , thanks for the explanation!

setting `.use_vulkan_debug_layer(false)` works.

is there a way to improve the error message? So that people that are new to vulkan and skulpin - like me :) - have a better getting started experience?

PS I am on fedora and installed `vulkan-validation-layers-devel` but it still wouldn't work with the debug layer on - I will update this if I figure out how to set it up.

**Comment by aclysma** [comment]
I think that error message is coming from vulkan, not skulpin so I can't change the text. (Aside from unpacking the error, string comparing, and returning something else instead.)

There is more info on installing the SDK here. Definitely recommended! https://github.com/aclysma/skulpin#requirements

I could use some bold text in a couple places in the readme to make it harder to miss and disable the debug layer by default.

**Comment by aclysma** [comment]
Oh there is also a TODO in the code for checking if the layer exists before trying to use it. That would also allow for clearer error handling.

**Comment by aclysma** [comment]
Fixed with #63, I'll release it as 0.10.0 if there aren't any issues found in the next few days.
